en/docs/install-and-setup/sso/okta-as-an-external-idp-using-saml.md

```diff
@@ -0,0 +1,163 @@
+
+    [![]({{base_path}}/assets/img/learn/okta-add-new-attribute-add.png)]({{base_path}}/assets/img/learn/okta-add-new-attribute-add.png) 
+
+6. Enter the following details and click save
+
+    [![]({{base_path}}/assets/img/learn/okta-add-new-attribute-details.png)]({{base_path}}/assets/img/learn/okta-add-new-attribute-details.png) 
+
+7. Go to Users -> People and click on your profile name. And navigate to the profile edit page as shown below
+
+    <img src="{{base_path}}/assets/img/learn/okta-profile-edit.png" />
+    <br/>
+    <br/>
+    <img src="{{base_path}}/assets/img/learn/okta-profile-edit2.png" width="600" height="400"/>
+    <br/>
+    <br/>
+
+    add the following role value. This will be used in the API-Manager to map an internal role to user that will be provisioned.
+    <img src="{{base_path}}/assets/img/learn/okta-profile-edit3.png"/>
+
+### Setup API-Manager
+1. Login in to `https://localhost:9443/carbon`.
+
+2. First we need to create a role that needs to be assinged to users that will be provisioned from okta. click on add in Users and Roles section and add a new role.
+    <img src="{{base_path}}/assets/img/learn/okta-apim-add-role.png" width="400" height="200"/>
+
+    <img src="{{base_path}}/assets/img/learn/okta-apim-add-role-name.png" width="400" height="200"/>
+
+    Assign the following permissions to the role and save
+
+    <img src="{{base_path}}/assets/img/learn/okta-apim-add-role-permissions3.png" width="300" height="300"/>
+    <br/>
+    <br/>
+    <img src="{{base_path}}/assets/img/learn/okta-apim-add-role-permissions2.png" width="300" height="350"/>
+    <br/>
+    <br/>
+    <img src="{{base_path}}/assets/img/learn/okta-apim-add-role-permissions1.png" width="300" height="300"/>
+
+3. Login to `https://localhost:9443/admin` expand settings & click on scope mapping
+
+    <img src="{{base_path}}/assets/img/learn/okta-apim-role-scope-mapping.png" width="300" height="500"/>
+
+    Update the following scopes with the okta_role
+
+    [![]({{base_path}}/assets/img/learn/okta-apim-role-scope-mapping-edit1.png)]({{base_path}}/assets/img/learn/okta-apim-role-scope-mapping-edit1.png) 
+    <br/>
+    <br/>
+    [![]({{base_path}}/assets/img/learn/okta-apim-role-scope-mapping-edit2.png)]({{base_path}}/assets/img/learn/okta-apim-role-scope-mapping-edit2.png) 
+    <br/>
+    <br/>
+    [![]({{base_path}}/assets/img/learn/okta-apim-role-scope-mapping-edit3.png)]({{base_path}}/assets/img/learn/okta-apim-role-scope-mapping-edit3.png) 
+
+    This will allow the user a user having the okta_role to login to Publisher and Developer Portal
+
+4. Login in to `https://localhost:9443/carbon` & Click on add in identity providers section. Enter Identity Provider Name. Expand the federated authenticators -> SAML2 Web SSO Configuration section and under Select Mode select Metadata File Configuration and update the xml file that was saved during the okta setup process.
```

Comment:
better if we highlight texts extracted from the UI.

Reply:
> highlight texts extracted from the UI.

Didn't get what did you mean here

Reply:
ex:
Click on **Add** in  **Identity Providers** section. Enter **Identity Provider Name** etc

Reply:
fixed https://github.com/wso2/docs-apim/pull/965/commits/2fc1ae7e318fc9d5bc274b7a3d09e86158228e7c